lib/kafka/consumer.rb

```diff
@@ -366,8 +366,12 @@ def mark_message_as_processed(message)
       @offset_manager.mark_as_processed(message.topic, message.partition, message.offset)
     end
 
-    def send_heartbeat_if_necessary
-      @heartbeat.send_if_necessary
+    def trigger_heartbeat
```

Comment:
Since this method has been part of the public API (Consumer is public) I don't want to change its name. We should either keep the old name as an alias or stick to the `send_*` naming at this layer.

Reply:
You are absolutely right. I will add proper public api references and we can just deprecate that at some point.

Reply:
@dasch added the compatibility layer.